xcode/Subconscious/Shared/Components/Settings/GatewayURLSettingsView.swift

```diff
@@ -47,17 +47,25 @@ struct ResourceSyncBadge: View {
 
 struct GatewayProvisionLabel: View {
     var status: ResourceStatus
+    var inviteCode: InviteCode?
+    var gatewayId: String?
     
     var label: String {
-        switch status {
-        case .initial:
-            return "Provision Gateway"
-        case .pending:
-            return "Provisioning..."
-        case .failed:
+        switch (status, inviteCode, gatewayId) {
+        case (.initial, .some(_), .some):
+            return "Check Gateway Status"
+        case (.pending, .some, .none):
+            return "Redeeming Invite Code..."
+        case (.pending, .some, .some):
+            return "Waiting..."
+        case (.failed, _, .none):
+            return "Failed to Redeem Invite Code"
+        case (.failed, _, .some):
             return "Provisioning Failed"
-        case .succeeded:
-            return "Gateway Created"
+        case (.succeeded, _, _):
+            return "Gateway Ready"
+        case _:
+            return "Provision Gateway"
```

Comment:
This may be excessive but I wanted to give more visible detail about the current state of the system. Makes me wonder if it's worth a bespoke `GatewayProvisioningStatus` enum?